scripts/asm-installer/vsphere/create_attach_context

```diff
@@ -0,0 +1,196 @@
+#!/bin/bash
+
+#set -ex
+set -CeE
+set -o pipefail
+set +o noclobber
+
+main() {
+  echo "============================================="
+  init
+  parse_validate_args "${@}"
+}
+
+init() {
+  # BSD-style readlink apparently doesn't have the same -f toggle on readlink
+  case "$(uname)" in
+  Linux) APATH="readlink" ;;
+  Darwin) APATH="stat" ;;
+  *) ;;
+  esac
+}
+
+parse_validate_args() {
+  # shellcheck disable=SC2064
+  trap "$(shopt -p nocasematch)" RETURN
+  shopt -s nocasematch
+
+  CONTEXT_FILE=".${1}"
+  read_args "${@}"
+  fill_unspecified_context
+  write_config_context
+
+}
+
+fill_unspecified_context() {
+
+  if [[ -z "${KUBECONFIG}" ]]; then
+    KUBECONFIG="$HOME/.kube/config"
+  fi
+  echo "Using KUBECONFIG ${KUBECONFIG}"
+
+  # use the absolute path for the key file
+  if [[ -f "${KEY_FILE}" ]]; then
+    KEY_FILE="$(apath -f "${KEY_FILE}")"
+    #    readonly KEY_FILE
+  elif [[ -n "${KEY_FILE}" ]]; then
+    fatal "Couldn't find key file ${KEY_FILE}. "
+  fi
+
+  CLUSTER_NAME=$(kubectl -n istio-system get IstioOperator installed-state --output=json | jq -r '.spec.values.global.multiCluster.clusterName')
+  if [[ ${CLUSTER_NAME} != ${CLUSTER} ]]; then
+    fatal "Current kubeconfig points to cluster: ${CLUSTER_NAME}, cannot connect to input cluster ${CLUSTER}"
+  fi
+
+
+  CLUSTER_NETWORK=$(kubectl -n istio-system get IstioOperator installed-state --output=json | jq -r '.spec.values.global.network')
+  if [[ ${CLUSTER_NETWORK} == ${VM_NETWORK} ]]; then
+    echo "Using Single-Network Scenario. Cluster and VM are Running in Network: ${VM_NETWORK}"
+  elif [[ -z ${VM_NETWORK} ]]; then
+    echo "VM Network      is not specified. Set VM Network same as Cluster Network ${CLUSTER_NETWORK}. Single-Network Scenario"
+    VM_NETWORK=${CLUSTER_NETWORK}
+  else
+    echo "Using Multi-Network Scenario. Cluster Network: ${CLUSTER_NETWORK}, VM Network: ${VM_NETWORK}"
+  fi
+
+  # name of the Kubernetes service account you want to use for your VM. same as "default" by default
+  if [[ -z "${SERVICE_ACCOUNT}" ]]; then
+    echo "Service Account is not specified. Set Service Account as default"
+    SERVICE_ACCOUNT="default"
+  fi
+
+  if [[ -z "${WORK_DIR}" ]]; then
+    WORK_DIR="$(pwd)"
+    echo "Working Dir     is not specified. Set Working Dir as ${WORK_DIR}"
+    echo "WORK_DIR=${WORK_DIR}" >>${CONTEXT_FILE}
+  fi
+
+  if [[ -z "${OUTPUT_DIR}" ]]; then
+    OUTPUT_DIR="${WORK_DIR}"
+    echo "Output  Dir     is not specified. Set Output  Dir as ${OUTPUT_DIR}/vm-vm_name"
```

Comment:
"Set Output  Dir as ${OUTPUT_DIR}/vm-vm_name". Do you mean $WORK_DIR/vm-vm_name?

Reply:
If user would like to save the generated artifacts somewhere other than work_dir, user can specify OUTPUT_DIR. If OUTPUT_DIR is unspecified, it will set as work_dir.